### PR TITLE
AK/FuzzyMatch: Return the best possible score for equality

### PR DESCRIPTION
**Pull request by LucasChollet**
This fixes cases where fuzzy matching would return a better score for
a different pattern than a needle perfectly matching the haystack.

As an example, when searching for "fire" in emojis:
 - "Fire Engine" would have scored 168
 - While "Fire" was only 160.
This patch makes the latter have the best possible score.

### CODE REVIEW

File: Tests/AK/TestFuzzyMatch.cpp [hunk 1]
```diff
@@ -60,11 +58,16 @@ TEST_CASE(is_separator_bonus_applied_correctly)
...
     EXPECT_EQ(fuzzy_match("abcd"sv, "a b c d"sv).score, 202);
     EXPECT_EQ(fuzzy_match("abcd"sv, "a_b_c_d"sv).score, 202);
     EXPECT_EQ(fuzzy_match("b c"sv, "ab cd"sv).score, 153);
     EXPECT_EQ(fuzzy_match("b_c"sv, "ab_cd"sv).score, 153);
     EXPECT_EQ(fuzzy_match("bc"sv, "ab cd"sv).score, 122);
 }
+
+TEST_CASE(equality)
+{
+
```
Comment (nico) [inline Tests/AK/TestFuzzyMatch.cpp:70]: weird newline

File: Tests/AK/TestFuzzyMatch.cpp [hunk 2]
```diff
@@ -60,11 +58,16 @@ TEST_CASE(is_separator_bonus_applied_correctly)
...
+    EXPECT_EQ(fuzzy_match("abc"sv, "abc"sv).score, NumericLimits<int>::max());
```
Comment (nico) [inline Tests/AK/TestFuzzyMatch.cpp:71]: Is it important that it's equal to `NumericLimits<int>::max()`? Or is just the line after it sufficient?

**Review by nico — COMMENTED** [review]
Thanks! lgtm, except for that one blank line. And one other question:

File: Tests/AK/TestFuzzyMatch.cpp (hunk 2 above)
Comment (LucasChollet) [reply Tests/AK/TestFuzzyMatch.cpp:70]: IMO result for equality should not be less than `NumericLimits<decltype(fuzzy_match(sv, sv))>::max()`. If it is less, there might be a non-equal string with a higher score. So this is the only case where the value actually matter.

Comment (nico) [reply Tests/AK/TestFuzzyMatch.cpp:70]: That's in theory possible for `NumericLimits<decltype(fuzzy_match(sv, sv))>::max()`, and in practice won't happen for many other values other than exactly `max()`, no?

Comment (nico) [reply Tests/AK/TestFuzzyMatch.cpp:70]: (To be clear, fine to keep as-is if you like it as-is.)

Comment (LucasChollet) [reply Tests/AK/TestFuzzyMatch.cpp:70]: True, the max value will still be an "issue", so it doesn't even add much theoretical value. I can't modify it before tonight though.

**Pull request merged**